qCC/ccGraphicalTransformationTool.cpp

```diff
@@ -210,6 +226,68 @@ bool ccGraphicalTransformationTool::setAdvancedTranslationTransform(ccHObject* t
 	}
 }
 
+bool ccGraphicalTransformationTool::setAdvancedRotationTransform(ccHObject* translateRef)
+{
+	if (!m_associatedWin)
+	{
+		assert(false);
+		return false;
+	}
+	if (translateRef->isA(CC_TYPES::POLY_LINE))
+	{
+		ccPolyline* line = static_cast<ccPolyline*>(translateRef);
+		CCVector3 start = *line->getPoint(0);
+		CCVector3 arbitraryVec = *line->getPoint(1) - *line->getPoint(0);
+		arbitraryVec.normalize();
+		m_advRotationAxis = m_rotation.inverse() * CCVector3d::fromArray(arbitraryVec.u);
+		if (refAxisRadio->isChecked())
+		{
+			ccGLMatrixd lineTransform = ccGLMatrixd(line->getGLTransformationHistory().data());
+			m_advRotationRefObjCenter = lineTransform.getTranslationAsVec3D();
+			ccLog::Print(QString("%1,%2,%3").arg(m_advRotationRefObjCenter[0]).arg(m_advRotationRefObjCenter[1]).arg(m_advRotationRefObjCenter[2]));
+			CCVector3d newCenter = (m_advRotationRefObjCenter - m_rotation.inverse() * m_position.getTranslationAsVec3D());
+			setRotationCenter(newCenter);
+		}
+		else
+		{
+			CCVector3d newCenter = CCVector3d::fromArray(m_toTransform.getBB_recursive().getCenter().u);
+			setRotationCenter(newCenter);
+		}
+		rotComboBox->setCurrentIndex(3); //Z
```

Comment:
Maybe use an enum here to be clearer? (and at the similar places as well)

Reply:
Sure, I can add an enum throughout